src/dev/run_prepush_hook.ts

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { existsSync } from 'node:fs';
+import { simpleGit } from 'simple-git';
+import cliProgress from 'cli-progress';
+import chalk from 'chalk';
+import Table from 'cli-table3';
+import { run } from '@kbn/dev-cli-runner';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { ToolingLog } from '@kbn/tooling-log';
+import { getTimeReporter } from '@kbn/ci-stats-reporter';
+
+import { File } from './file';
+import { checkFileCasing } from './preflight_check/check_file_casing';
+import { checkTypescriptFiles } from './preflight_check/run_tsc';
+import { runUnitTests } from './preflight_check/run_unit_tests';
+import { esLintFiles } from './preflight_check/run_eslint';
+
+const HASH_FOR_NULL_OR_DELETED_FILE = '00000000000';
+
+function getDefaults(taskName: string) {
+  const TASK_COL_LENGTH = 10;
+  return {
+    task: chalk.blue(taskName.padEnd(TASK_COL_LENGTH)),
+    filename: '',
+  };
+}
+
+function nonNullable<T>(value: T): value is NonNullable<T> {
+  return value != null;
+}
+
+run(
+  async ({ log, flags }) => {
+    const toolingLog = new ToolingLog({
+      level: 'info',
+      writeTo: process.stdout,
+    });
+
+    const runStartTime = Date.now();
+    const reportTime = getTimeReporter(toolingLog, 'scripts/run_prepush_hook');
+
+    process.env.IS_KIBANA_PREPUSH_HOOK = 'true';
+
+    const git = simpleGit(REPO_ROOT);
+
+    const checkTypes = ['i18n', 'tsc', 'eslint', 'jest', 'fileCasing'] as const;
+
+    const checks = checkTypes.reduce((acc, check) => {
+      acc[check] = {
+        files: [],
+      };
+
+      return acc;
+    }, {} as Record<typeof checkTypes[number], { files: Array<{ path: string; file: File }> }>);
+
+    const { current } = await git.branchLocal();
+
+    const commonAncestor = (await git.raw(['merge-base', 'origin/main', current])).trim();
+
+    const diff = await git.diff([`${commonAncestor}..${current}`]);
+
+    const filesChangedSummaryTable = new Table({
+      head: ['Files changed in this PR compared to upstream:', 'Status'],
```

Comment:
Just small word changes to be a bit more specific:

```suggestion
      head: ['Files changed in this branch compared to origin/main:', 'Status'],
```